sdk/servicebus/azure_messaging_servicebus/src/receiver.rs

```diff
@@ -0,0 +1,3758 @@
+//!     max_message_count: 10,
+//!     max_wait_time: Some(Duration::seconds(30)),
+//! };
+//!
+//! let messages = receiver.receive_messages(10, Some(options)).await?;
+//! for message in &messages {
+//!     println!("Received: {:?}", String::from_utf8_lossy(message.body()));
+//!     receiver.complete_message(message, None).await?;
+//! }
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! ## Error Handling with Settlement
+//!
+//! ```rust,no_run
+//! use azure_messaging_servicebus::{
+//!     ServiceBusClient,
+//!     ReceiveMessageOptions, DeadLetterMessageOptions, AbandonMessageOptions
+//! , ServiceBusClientOptions};
+//!
+//! use azure_identity::DeveloperToolsCredential;
+//! # #[tokio::main]
+//! # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let credential = DeveloperToolsCredential::new(None)?;
+//! let client = ServiceBusClient::builder().open("myservicebus.servicebus.windows.net", credential.clone()).await?;
+//! let receiver = client.create_receiver("my-queue", None).await?;
+//!
+//! if let Some(message) = receiver.receive_message(None).await? {
+//!     match process_message(&message).await {
+//!         Ok(_) => {
+//!             receiver.complete_message(&message, None).await?;
+//!         }
+//!         Err(ProcessingError::Retryable) => {
+//!             receiver.abandon_message(&message, None).await?;
+//!         }
+//!         Err(ProcessingError::Fatal) => {
+//!             let dead_letter_options = DeadLetterMessageOptions {
+//!                 reason: Some("ProcessingFailed".to_string()),
+//!                 error_description: Some("Unrecoverable processing error".to_string()),
+//!                 properties_to_modify: None,
+//!             };
+//!             receiver.dead_letter_message(&message, Some(dead_letter_options)).await?;
+//!         }
+//!     }
+//! }
+//! # Ok(())
+//! # }
+//! # async fn process_message(message: &azure_messaging_servicebus::ReceivedMessage) -> Result<(), ProcessingError> { Ok(()) }
+//! # #[derive(Debug)] enum ProcessingError { Retryable, Fatal }
+//! ```
+
+use crate::{
+    client::ServiceBusClientOptions, message::SystemProperties, ErrorKind, ReceivedMessage, Result,
+    ServiceBusError,
+};
+use async_lock::{Mutex, OnceCell};
+use azure_core::{fmt::SafeDebug, time::Duration, time::OffsetDateTime, Uuid};
+use azure_core_amqp::{
+    message::{AmqpMessageBody, AmqpMessageId},
+    AmqpConnection, AmqpDelivery, AmqpDeliveryApis, AmqpManagementApis, AmqpReceiver,
+    AmqpReceiverApis, AmqpSession, AmqpSessionApis, AmqpSimpleValue, AmqpSource, AmqpValue,
+};
+use futures::{select, FutureExt};
+use std::{collections::HashMap, sync::Arc};
+use tracing::{debug, trace, warn};
+
+/// Represents the lock style to use for a receiver - either `PeekLock` or `ReceiveAndDelete`.
+///
+/// This enum controls when a message is deleted from Service Bus and determines how message
+/// settlement works for received messages.
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
Enums are still models and should derive `SafeDebug`. See previous comment for more info on why.